back-end/app/Http/Controllers/User/UsersController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace App\Http\Controllers\User;
+
+use App\Models\User;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Hash;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class UsersController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $users = User::where('is_admin', false)->get();
+
+        return response()->json([
+            'data' => $users,
+        ], 200);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+
+        $user = $request->validate([
+            'first_name' => 'required|max:191',
+            'last_name' => 'required|max:191',
+            'email' => 'required|unique:users|max:191',
+            'password' => 'required',
+        ]);
+
+        User::create([
+            'first_name' => $user['first_name'],
+            'last_name' => $user['last_name'],
+            'email' => $user['email'],
+            'password' => bcrypt($user['password']),
+        ]);
+
+
+        return response()->json([
+            "message" => "Account Created Successfully",
+        ], 201);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($user_id)
+    {
+        $user = User::findOrFail($user_id);
+
+        if ($user->is_admin) throw new HttpException(404, 'User not found');
+
+        return response()->json([
+            'data' => $user,
+        ], 200);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $user_id)
+    {
+        $user = $request->validate([
+            'first_name' => 'required|max:191',
+            'last_name' => 'required|max:191',
+            'email' => 'required|unique:users|max:191',
+            'password' => 'required',
+        ]);
+
+        User::findOrFail($user_id)->update([
+            'first_name' => $user['first_name'],
+            'last_name' => $user['last_name'],
+            'email' => $user['email'],
+            'password' => bcrypt($user['password']),
+        ]);
+
+        return response()->json([
+            "message" => "Account Updated Successfully",
+        ], 201);
+    }
```

Comment:
I don't think we need this one, especially  the password update but I will leave as it is for now.

Reply:
Regarding this, sir, I was thinking that we could use it for the admin when updating a user's information. I will update this section as we progress in developing the app to be more detailed.